src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6567,13 +6567,19 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 ASR::ttype_t* arg_type = nullptr;
                 if( ASR::is_a<ASR::Var_t>(*args[0].m_value) ) {
                     ASR::Var_t* arg_Var = ASR::down_cast<ASR::Var_t>(args[0].m_value);
-                    if( ASR::is_a<ASR::Variable_t>(*arg_Var->m_v) ) {
-                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var->m_v)->m_type;
-                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var->m_v) ) {
+                    ASR::symbol_t* arg_Var_m_v = ASRUtils::symbol_get_past_external(arg_Var->m_v);
+                    if( ASR::is_a<ASR::Variable_t>(*arg_Var_m_v) ) {
+                        // TODO: Import the underlying struct if arg_type is of Struct type
+                        // Ideally if a variable of struct type is being imported then its underlying type
+                        // should also be imported automatically. However, the naming of the
+                        // underlying struct type might lead to collisions, so importing the type
+                        // here seems like a better choice. Should be done later when the case arises.
+                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var_m_v)->m_type;
+                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var_m_v) ) {
```

Comment:
Based on my previous comment, this shouldn't even pass verify(). I guess it does in LPython, but will not once we sync ASR with LFortran, where I improved the pass. But we'll simply allow Struct as well in verify.

Reply:
Could you please share the PR where you improved the ASR verify pass? I will apply those changes and test right away.

Reply:
Hm, another question: what type is the Var, or the sizeof argument, if it points to Struct (=StructType currently)? The type is the "struct class itself, not an instance". After refactor, for the instance of a Struct has a type StructType. But what type does Struct itself have?